recipes/hdf5/all/conanfile.py

```diff
@@ -229,14 +241,16 @@ def _config_libname(lib):
 
             self.cpp_info.components[component_name].set_property("cmake_target_name", f"hdf5::{alias_target}")
             self.cpp_info.components[component_name].set_property("pkg_config_name", alias_target)
+            self.cpp_info.components[component_name].set_property("cmake_build_modules", [self._module_file_variables_rel_path])
             self.cpp_info.components[component_name].libs = [_config_libname(alias_target)]
             self.cpp_info.components[component_name].requires = requirements
+            self.cpp_info.components[component_name].builddirs = [os.path.join("lib", "cmake")]
```

Comment:
```suggestion
```

Reply:
hook is wrong, it's useless.

Reply:
Ok thanks @SpaceIm ,
please see latest commit to change it to append,
however, when you say the "hook is useless", do you mean I should not be appending to builddirs?
Or is this ok now?

Reply:
You should not append nor override here, the hook is wrong.

Reply:
I do have a 2 cmake_build_modules file that is generated ...
ah maybe because I didn't include it in that set_property("cmake_build_modules") line 245...